docs/typescript.md

```diff
@@ -1,14 +1,170 @@
-# TypeScript Guidelines
+# **Typescript guidelines**
 
-These guidelines specifically apply to TypeScript.
+**_Purpose_**: Establish best practices for writing robust, maintainable, and efficient TypeScript code.
 
-## Provide explicit return types for functions/methods
+**_Audience_**: Internal and external developers using TypeScript in Metamask.
+
+## **Interfaces**
+
+By defining expected object structures with interfaces, TypeScript promotes developer intent and unlocks powerful IDE features like accurate completion and context-aware navigation.
```

Comment:
FYI: We're enforcing usage of type aliases over interfaces in all of our packages with the eslint rule `@typescript-eslint/consistent-type-definitions`.

See the following for previous discussions on this matter:
- https://github.com/MetaMask/eslint-config/pull/216
- https://github.com/MetaMask/core/pull/1933

Reply:
Will change the title to `type`, thanks for the heads up

Reply:
Continued in https://github.com/MetaMask/contributor-docs/pull/80#discussion_r1506955448